[http3] Enable tests for corrupted 2nd initial

hfujita opened the pull request:
This is a follow-up PR for #2614.

* remove Python 3 requirement so the test run with the current Docker image
* enabled one test which was accidentally commented out

kazuho commented:
Thank you for the PR. Would you be interested in (or would you mind me) rewriting udp-generator.py in perl? Doing so would eliminate the dependency problem.

hfujita commented:
@kazuho done in 57c8f8d. Please take a look, thank you!

kazuho commented on t/udp-generator.pl:
```diff
@@ -0,0 +1,34 @@
+
+# Read datagrams from input files and send them to a specified server
+
+use strict;
```
```suggestion
use strict;
use warnings;
```

Generally speaking it's good to enable both strict and warning.

kazuho commented on t/udp-generator.pl:
```diff
@@ -0,0 +1,34 @@
+use IO::File;
+use IO::Socket;
+
+if (scalar @ARGV < 3) {
```
```suggestion
if (@ARGV < 3) {
```

"scalar" can be omitted when comparing against a scalar.

kazuho commented on t/udp-generator.pl:
```diff
@@ -0,0 +1,34 @@
+) or die "Failed to create a UDP socket: $!\n";
+
+foreach my $f (@ARGV) {
+	my $fh = IO::File->new($f, "r") or die "Failed to open file: $!\n";
```
```suggestion
	my $fh = IO::File->new($f, "r") or die "Failed to open file: $f:$!";
```
Maybe print the filename as well?

Also, generally speaking it's good to omit newline at the end of `die` or `warn`, because then perl would emit the location within the source code at where the problem occurred. That applies to other parts using `die` as well.

hfujita replied:
Fixed in 9a355c6 and 65f477e.

kazuho commented on t/udp-generator.pl:
```diff
@@ -0,0 +1,34 @@
+	PeerPort => $port,
+) or die "Failed to create a UDP socket: $!\n";
+
+foreach my $f (@ARGV) {
```
```suggestion
for my $f (@ARGV) {
```

"foreach" is a synonym of "for" and is rarely used today.

kazuho reviewed (commented):
@hfujita Sorry for the delay.

Thank you very much for your efforts porting the script to perl. I understand that perl is something that you might not want to spend too much time.

The code looks fine, I just have stylish comments.

hfujita commented:
@kazuho Thank you for the review. I think I have addressed all of your comments.

 It was a great Perl learning opportunity for me. I deeply appreciate your patience! 😄

kazuho commented:
Thank you!!!
